server/protocols/webadmin/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java

```diff
@@ -0,0 +1,126 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.user.api.model.User;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.model.AddUserRequest;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+public class UserRoutes implements Routes {
+
+    private static final String USER_NAME = ":userName";
+    private static final String EMPTY_BODY = "";
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserRoutes.class);
+
+    public static final String USERS = "/users";
+
+    private final UsersRepository usersRepository;
+    private final JsonTransformer jsonTransformer;
+    private final JsonExtractor<AddUserRequest> jsonExtractor;
+
+    @Inject
+    public UserRoutes(UsersRepository usersRepository, JsonTransformer jsonTransformer) {
+        this.usersRepository = usersRepository;
+        this.jsonTransformer = jsonTransformer;
+        this.jsonExtractor = new JsonExtractor<>(AddUserRequest.class);
+    }
+
+    @Override
+    public void define(Service service) {
+        service.get(USERS,
+            (request, response) -> getUsers(),
+            jsonTransformer);
+
+        service.post(USERS, this::addUser);
```

Comment:
if username is an identifier for delete, it must also be for creation. You should use PUT here too